addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -456,6 +512,17 @@ function factory(dependencies) {
                 'messagesAreEmpty',
             ],
         }),
+        /**
+         * Handles the message_fetch retries.
+         * This is not a "real" field, its compute function is used to re-trigger
+         * the message fetch after some time.
+         */
+        onHasLoadingFailedChanged: attr({
```

Comment:
Just a comment about something that looks off when I look at recent changes, but don't consider it in this PR:

I see many newly introduced `onChange` compute methods. I don't really like this... this puts a lot of side-effects, and I feel it's way overused. Also this is a workaround with computed fields and explicit dependencies, so it's not something that was intentionally designed to work like that...
Sure we see some need to "watch" stuffs, but parts that need them don't yet have this feature, and the `onChange` ones look "lazy" way of implementing something. It's not as bad as `_updateAfter`, but it shares some problems...

Reply:
This one has been removed, tough